ext/zpages/test/tracez_data_aggregator_benchmark.cc

```diff
@@ -0,0 +1,249 @@
+#include <chrono>
+
+#include "opentelemetry/sdk/trace/tracer.h"
+
+using namespace opentelemetry::sdk::trace;
+using namespace opentelemetry::ext::zpages;
+using opentelemetry::core::SteadyTimestamp;
+
+/////////////////////////////// BENCHMARK HELPER FUNCTIONS //////////////////////////////
+
+/*
+ * Helper function that creates and ends i spans instantly.
+ */
+void StartEndSpans(
+    std::shared_ptr<opentelemetry::trace::Tracer> &tracer,
+    int i,
+    bool isUnique = false)
+{
+  for (; i > 0; i--)
+    tracer->StartSpan(isUnique ? std::to_string(i) : "")->End();
+}
+
+/*
+ * Helper function that creates and ends spans instantly, while simulating
+ * different latencies.
+ */
+void StartEndSpansLatency(
+    std::shared_ptr<opentelemetry::trace::Tracer> &tracer,
+    int i,
+    bool isUnique = false)
+{
+  opentelemetry::trace::StartSpanOptions start;
+  start.start_steady_time = SteadyTimestamp(nanoseconds(0));
+  for (; i > 0; i--)
+  {
+    // Latency bucket depends on the index
+    auto latency_band = kLatencyBoundaries[i % kLatencyBoundaries.size()];
+    opentelemetry::trace::EndSpanOptions end;
+    end.end_steady_time = SteadyTimestamp(latency_band);
+
+    tracer->StartSpan(isUnique ? std::to_string(i) : "", start)->End(end);
+  }
+}
+/*
+ * Helper function that creates and ends spans instantly, while simulating
+ * error codes.
+ */
+void StartEndSpansError(
+    std::shared_ptr<opentelemetry::trace::Tracer> &tracer,
+    int i,
+    bool isUnique = false)
+{
+  for (; i > 0; i--)
+    tracer->StartSpan(isUnique ? std::to_string(i) : "")
+        ->SetStatus(opentelemetry::trace::CanonicalCode::CANCELLED, "");
+}
+
+/*
+ * Helper function that creates i spans that always run.
+ */
+void StartSpans(
+    std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>> &spans,
+    std::shared_ptr<opentelemetry::trace::Tracer> &tracer,
+    int i,
+    bool isUnique = false)
+{
+  for (; i > 0; i--)
+    spans.push_back(tracer->StartSpan(isUnique ? std::to_string(i) : ""));
+}
+
+/*
+ * Helper function that queries for Tracez data. This simulates a user visiting
+ * or refreshing the Tracez webpage many times, except this function does and
+ * holds nothing in memory. While this runs, aggregation work and performance is
+ * expected is be affected negatively.
+ */
+
+void GetManyAggregations(std::unique_ptr<TracezDataAggregator> &aggregator, int i)
+{
+  for (; i > 0; i--)
+    aggregator->GetAggregatedTracezData();
+}
+
+////////////////////////  FIXTURE FOR SHARED SETUP CODE ///////////////////
+
+class TracezAggregator : public benchmark::Fixture
+{
+protected:
+  void SetUp(const ::benchmark::State& state)
+  {
+    std::shared_ptr<TracezSpanProcessor> processor(new TracezSpanProcessor());
+    tracer     = std::shared_ptr<opentelemetry::trace::Tracer>(new Tracer(processor));
+    aggregator = std::unique_ptr<TracezDataAggregator>(
+        new TracezDataAggregator(processor, milliseconds(10)));
```

Comment:
10 milliseconds can be another param for the benchmark. 
We can also keep it as is for the first iteration, and then maybe it worth to add a comment that explains what is the number in production and why here in the benchmark it is lower.

Reply:
do you think i need only 1 other param for periodic fetches? i'm thinking 10 nanoseconds?

Reply:
One should be good as a start. Exposing it as a param gives the flexibility to add additional values if needed.